api/src/org/apache/cloudstack/api/command/user/vm/UpgradeVMCmd.java

```diff
@@ -77,7 +80,18 @@ public Long getServiceOfferingId() {
     }
 
     public Map<String, String> getDetails() {
-        return details;
+        Map<String, String> customparameterMap = new HashMap<String, String>();
+        if (details != null && details.size() != 0) {
+            Collection parameterCollection = details.values();
+            Iterator iter = parameterCollection.iterator();
+            while (iter.hasNext()) {
+                HashMap<String, String> value = (HashMap<String, String>)iter.next();
```

Comment:
the findbugs issue is here, a string is returned as a HashMap<String, String>. The fact that this works is very ugly. It means that details is really a Map<String,Map<String,String>>. If we want to both revert the issue and solve the findsbugs issue, the solution might be something like writing a parser and parse the string iter.next() into a map. Not sure  though